fe/fe-core/src/main/java/org/apache/doris/policy/RowPolicy.java

```diff
@@ -0,0 +1,174 @@
+import org.apache.doris.catalog.Database;
+import org.apache.doris.catalog.Table;
+import org.apache.doris.common.AnalysisException;
+import org.apache.doris.common.io.Text;
+import org.apache.doris.common.util.SqlParserUtils;
+import org.apache.doris.persist.gson.GsonUtils;
+
+import com.google.common.collect.Lists;
+import com.google.gson.annotations.SerializedName;
+import lombok.Data;
+import org.apache.commons.lang.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.DataInput;
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.List;
+
+/**
+ * Save policy for filtering data.
+ **/
+@Data
+public class RowPolicy extends Policy {
+
+    private static final Logger LOG = LogManager.getLogger(RowPolicy.class);
+
+    /**
+     * Policy bind user.
+     **/
+    @SerializedName(value = "user")
+    private UserIdentity user = null;
+
+    @SerializedName(value = "dbId")
+    private long dbId = -1;
+
+    @SerializedName(value = "tableId")
+    private long tableId = -1;
+
+    /**
+     * PERMISSIVE | RESTRICTIVE, If multiple types exist, the last type prevails.
+     **/
+    @SerializedName(value = "filterType")
+    private FilterType filterType = null;
+
+    /**
+     * Use for Serialization/deserialization.
+     **/
+    @SerializedName(value = "originStmt")
+    private String originStmt;
+
+    private Expr wherePredicate = null;
+
+    public RowPolicy() {}
+
+    /**
+     * Policy for Table. Policy of ROW or others.
+     *
+     * @param type PolicyType
+     * @param policyName policy name
+     * @param dbId database i
+     * @param user username
+     * @param originStmt origin stmt
+     * @param tableId table id
+     * @param filterType filter type
+     * @param wherePredicate where predicate
+     */
+    public RowPolicy(final PolicyTypeEnum type, final String policyName, long dbId,
+                     UserIdentity user, String originStmt, final long tableId,
+                     final FilterType filterType, final Expr wherePredicate) {
+        super(type, policyName);
+        this.user = user;
+        this.dbId = dbId;
+        this.tableId = tableId;
+        this.filterType = filterType;
+        this.originStmt = originStmt;
+        this.wherePredicate = wherePredicate;
+    }
+
+    /**
+     * Use for SHOW POLICY.
+     **/
+    public List<String> getShowInfo() throws AnalysisException {
+        Database database = Catalog.getCurrentCatalog().getDbOrAnalysisException(this.dbId);
+        Table table = database.getTableOrAnalysisException(this.tableId);
+        return Lists.newArrayList(this.policyName, database.getFullName(), table.getName(), this.type.name(),
+                this.filterType.name(), this.wherePredicate.toSql(), this.user.getQualifiedUser(), this.originStmt);
+    }
+
+    /**
+     * Read Table Policy from file.
+     **/
+    public static RowPolicy read(DataInput in) throws IOException {
```

Comment:
This `read` method should be implement in `Policy` abstract class.
And you need to add a `RuntimeTypeAdapterFactory` in `GsonUtils` for derived class.
You can refer to `Resource.java` and its derived class like `SparkResource` and `S3Resource`.

And please and ut for the read/write mothod of RowPolicy, you can refer to `DropInfoTest.java`